### PR TITLE
Always sign transactions and messages with the correct account

### PR DESCRIPTION
**Pull request by Gudahtt**
The account used to sign transactions and messages should be the one the transaction or message is from. Instead, the last connected account was being used to sign any messages or transactions.

This was especially problematic considering the last connected account was not persisted, meaning that signatures were being performed with the wrong account after a reset (unless the last connected account happened to be account 0, which was the default).

A mapping of addresses to indexes as been added to the keyring state, and this mapping has been persisted. This should ensure the correct account index is used, and thus the correct hd path, each time this keyring is used for signing.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -45,6 +47,7 @@ class LedgerBridgeKeyring extends EventEmitter {
     this.hdPath = opts.hdPath || hdPathString
     this.bridgeUrl = opts.bridgeUrl || BRIDGE_URL
     this.accounts = opts.accounts || []
+    this.accountIndexes = opts.accountIndexes || {}
```
Comment (whymarrh) [inline index.js]: I think we can drop the fallback here:

```suggestion
    this.accountIndexes = opts.accountIndexes
```

**Comment by whymarrh** [comment]
I can confirm that this does indeed sign transactions from *n*th accounts correctly, where _n > 1_

File: index.js (hunk 1 above)
Comment (Gudahtt) [reply index.js]: Can we? If `this.accountIndexes` is undefined, then various other lines added in this PR will fail.

Comment (whymarrh) [reply index.js]: Good point, this suggestion doesn't account for deserializing an old instance

**Comment by whymarrh** [comment]
The reproduction steps one can follow in MetaMask:

**Before:**

1. Connect a Ledger account > 1 (e.g. the 2nd account)
2. Confirm that sending a transaction is signed correctly
3. Reload MetaMask (or, for non-MM, construct a new instance from serialized state)
4. Confirm that sending a transaction is signed with the 1st account on the Ledger (an incorrect account)

**After:**

1. Connect a Ledger account > 1 (e.g. the 2nd account)
2. Confirm that sending a transaction is signed correctly
3. Reload MetaMask (or, for non-MM, construct a new instance from serialized state)
4. Confirm that sending a transaction is signed with the 2nd account on the Ledger (the correct account)

**Pull request merged**